pkg/tfgen/generate.go

```diff
@@ -373,6 +373,11 @@ func (v *variable) optional() bool {
 		return true
 	}
 
+	//if we have an explicit marked as optional then let's return that
+	if v.info != nil && v.info.MarkAsOptional != nil {
```

Comment:
Minor - but looks like we aren't actually checking the value of the bool?  Humorously, this is a case of confusing ternary logic similar to the one this issue is addressing :-).